packages/core/src/RadioCheckToggleCard/RadioCheckToggleCard.tsx

```diff
@@ -0,0 +1,216 @@
+    --icon-border-shadow-color: ${getPaletteColor('background.light')};
+  }
+
+  &:active + div {
+    --border-color: ${getPaletteColor('primary.dark')};
+    --icon-color: ${getPaletteColor('primary.dark')};
+  }
+
+  &:checked + div {
+    --background-color: ${getPaletteColor('primary.light')};
+    --border-color: ${getPaletteColor('primary.base')};
+    --icon-color: ${getPaletteColor('primary.base')};
+  }
+
+  &:checked:hover + div {
+    --border-color: ${getPaletteColor('primary.dark')};
+    --icon-color: ${getPaletteColor('primary.dark')};
+  }
+
+  &:checked:focus-visible + div {
+    --icon-border-color: ${getPaletteColor('primary.dark')};
+  }
+
+  &:disabled + div {
+    color: ${getPaletteColor('border.base')};
+  }
+`
+
+const RCTCardContainer = styled(Flex)`
+  --background-color: ${getPaletteColor('background.lightest')};
+  --border-color: transparent;
+  --default-card-margin: ${themeGet('space.3')};
+  --icon-color: ${getPaletteColor('text.light')};
+  --icon-border-color: transparent;
+  --icon-border-shadow-color: transparent;
+
+  background-color: var(--background-color);
+  border: 1px solid var(--border-color);
+  border-radius: ${themeGet('borderRadii.action-lg')};
+  box-shadow: ${themeGet('shadows.sm')};
+  cursor: pointer;
+  padding: var(--default-card-margin);
+`
+
+const CardHeader = styled(Flex)`
+  min-width: 0;
+  width: 100%;
+
+  & > svg {
+    border: 1px solid var(--icon-border-color);
+    border-radius: var(--icon-border-radius);
+    color: var(--icon-color);
+    padding: 2px;
+    box-shadow: inset 0 0 0 5px var(--icon-border-shadow-color);
+  }
+
+  & + * {
+    margin-top: ${themeGet('space.3')};
+  }
+`
+
+const CardContent = styled(Box)`
+  flex: 1;
+
+  & > :first-child + * {
+    margin-top: ${themeGet('space.3')};
+  }
+`
+
+const RadioIcon = ({ checked, ...props }) => {
+  return checked ? <RadioChecked {...props} /> : <RadioEmpty {...props} />
+}
+
+const CheckboxIcon = ({ checked, ...props }) => {
+  return checked ? <BoxChecked {...props} /> : <BoxEmpty {...props} />
+}
+
+const ToggleIcon = ({ checked, onChange, ...props }) => {
+  const checkIcon = <Check size='16' />
+  return (
+    <Box {...props}>
+      <Toggle height={24} isOn={checked} icon={checkIcon} onToggle={onChange} />
+    </Box>
+  )
+}
+
+const buttonIcon = (cardType: TCardTypes) => {
+  if (cardType === 'radio') {
+    return RadioIcon
+  } else if (cardType === 'checkbox') {
+    return CheckboxIcon
+  } else if (cardType === 'toggle') {
+    return ToggleIcon
+  }
+}
+
+const RadioCheckToggleCard = (props: IRadioCheckToggleCard) => {
+  const {
+    children,
+    cardType,
+    hPosition,
+    vPosition,
+    isHeightOccupied,
+    title,
+    isTitleBold,
+    name,
+    value,
+    onChange,
+    isSelected,
+  } = props
+
+  const id = `${name}_${value}`
+  const DisplayButton = buttonIcon(cardType)
+
+  const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => onChange(e)
```

Comment:
`onChange` isn't a required prop, right? I think this should have a conditional only to execute `onChange` if it's present.